config/taxonomy/base/base.yaml

```diff
@@ -31,3 +31,36 @@ definitions:
       dataformat:
         $ref: "#/definitions/Dataformat"
     additionalProperties: false
+  Source:
```

Comment:
Shouldn't the taxonomy include enums for the plugin types, dependency types, capability scope, and the supported actions?

Reply:
@simanadler can you explain what should be defined in the CRD and what should be defined in the taxonomy.
Keep in mind that the taxonomy target cross components definitions. Are these definitions referenced beyond the module definition?
For actions, they are not part of the base, but a sample, as they are deployment specific.

Reply:
@eletonia @roee88 @ronenkat The following are the constants in fybrikmodule_types.go and my thoughts about if/why they should be in the taxonomy:

CapabilityScope - (asset, workload, cluster) - If we think that these will not change or that we are happy with creating a new fybrik release if they change, then this can remain a constant and not go in the taxonomy.

DependencyType - same reasoning as CapabilityScope

CapabilityType (read, write, read, transform) - As new types of capabilities and modules are created additional values will likely need to be added.  Ex: isolation, data quality, logging, etc.  Thus, these should be in the taxonomy in my opinion.

PluginMechanism - should be in the taxonomy, so that new types of plugins  can be added without having to upgrade to a new fybrik version.

Actions - definitely need to be in the taxonomy.  If we provide out of the box actions in the quick start (or other pre-configured deployments), those actions should be included in the taxonomy provided with the quickstart.  Today the data movement operator is included in the fybrik repository, so shouldn't the actions it support be in the default taxonomy?

Data Protocols and Formats - don't know whether these should be in the data catalog or module taxonomy, but they need to be in one of them.  Again, those protocols and formats supported by the fybrik basic installation should be included in the taxonomy in my opinion.

Module Type - Currently a string whose documentation says it should be a service, config or plugin.  If we think that these will not change or that we are happy with creating a new fybrik release if they change, then I guess this doesn't need to go in the taxonomy.

Reply:
@simanadler for everything you wrote as "if  happy to create a new release" I would go for excluding it from the taxonomy until we decide otherwise. 

With that in mind, I think that we're already aligned with the exception of CapabilityType which I'm unsure about (discussion needed; for this PR my suggestion was to exclude it and raise this in an open source meeting).

Reply:
@simanadler @roee88 Seems like the only fields we would potentially add to the current taxonomy are `PluginMechanism` and `CapabilityType`
I don't see `PluginMechanism` in any of the module specs - neither in the [arrow flight module](https://github.com/fybrik/arrow-flight-module/blob/master/module.yaml) nor the [hello world read module](https://github.com/fybrik/hello-world-read-module/blob/main/hello-world-read-module.yaml). 

Does this mean `PluginMechanism` is optional and not required? What are the possible values for `PluginMechanism`?

Reply:
@ronenkat Actually we already have `CapabilityType` in the taxonomy as an enum [here](https://github.com/fybrik/fybrik/blob/8d62b60521c9005b15ac067470e4172977528af0/config/taxonomy/base/base.yaml#L47) @roee88 Why do you think we should exclude it?